.github/workflows/lintercheck.yml

```diff
@@ -0,0 +1,79 @@
+name: Linter check
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags:
+      - r[0-9]+.[0-9]+
+
+jobs:
+  linter_check:
+    runs-on: ubuntu-20.04
```

Comment:
why not ubuntu-latest?

Reply:
Because there are no `clang-format-7` in `ubuntu-22:04` (aka `ubuntu-latest`) and `clang-format-14` suggests a lots of changes